### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by Thermiee**
Project requirements for this pull request:

- Remove the current_user method in ApplicationController, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
- Can register a new user.
- The user logs in with a combination of email and password.
- Hashed passwords should be stored in the database.
- Ask for confirmation of email.
- Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable,
+         :confirmable
   has_many :posts, foreign_key: 'author_id'
```
Comment (sja-thedude) [inline app/models/user.rb]: - [ ] Great job so far, however, I cant signup or login into your app. When I tried signing up the user id rollback as shown below. And when I tried to log in it doesn't log in and when I checked User.all the users are not displayed there.

https://user-images.githubusercontent.com/70269507/173555325-a0ccb0bb-0321-46b0-8f76-8b3438d4b856.mov

https://user-images.githubusercontent.com/70269507/173555364-e878458d-6889-4a67-bbee-5ad4fc64e42e.mov

For resolving the current issue of the roll back add this line html: {'data-turbo' => "false"}, to the form_for tag in the new.html.erb file

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @cynthiainga, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Good job so far
✔️ Good PR title and description
✔️ All linters passing

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lameck1 — APPROVED** [review]
Hi, @cynthiainga,

Tasks for this project have been successfully implemented 💯.

## Highlights
- Professional README ✔️
- Clean code ✔️ 
- Good commit messages ✔️ 

All the best with the next project.

You are free to merge this branch :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**Please, remember to tag me in your question so I can receive the notification.**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by omar-labana** [comment]
@Thermiee why are you asking for another CRR? did the previous reviewer mark it as changes requested?

**Comment by Thermiee** [comment]
> @Thermiee why are you asking for another CRR? did the previous reviewer mark it as changes requested?

@omar-labana Microverse left the request review on my dashboard but it has been resolved

**Pull request merged**